RecoEgamma/EgammaElectronProducers/plugins/GsfElectronBaseProducer.cc

```diff
@@ -22,131 +21,238 @@
+  desc.add<double>("minEOverPEndcaps", 0.0);
+  desc.add<double>("maxEOverPEndcaps", 999999999.0);
+  desc.add<double>("minEOverPBarrel", 0.0);
+  desc.add<double>("maxEOverPBarrel", 999999999.0);
+  desc.add<double>("maxDeltaEtaBarrel", 0.02);
+  desc.add<double>("maxDeltaEtaEndcaps", 0.02);
+  desc.add<double>("maxDeltaPhiBarrel", 0.15);
+  desc.add<double>("maxDeltaPhiEndcaps", 0.15);
+  desc.add<double>("hOverEConeSize", 0.15);
+  desc.add<double>("hOverEPtMin", 0.0);
+  desc.add<double>("maxHOverEBarrel", 0.15);
+  desc.add<double>("maxHOverEEndcaps", 0.15);
+  desc.add<double>("maxHBarrel", 0.0);
+  desc.add<double>("maxHEndcaps", 0.0);
+  desc.add<double>("maxSigmaIetaIetaBarrel", 999999999.0);
+  desc.add<double>("maxSigmaIetaIetaEndcaps", 999999999.0);
+  desc.add<double>("maxFbremBarrel", 999999999.0);
+  desc.add<double>("maxFbremEndcaps", 999999999.0);
+  desc.add<bool>("isBarrel", false);
+  desc.add<bool>("isEndcaps", false);
+  desc.add<bool>("isFiducial", false);
+  desc.add<bool>("seedFromTEC", true);
+  desc.add<double>("maxTIP", 999999999.0);
+
+  // preselection parameters (tracker driven only electrons)
+  desc.add<double>("minSCEtBarrelPflow", 0.0);
+  desc.add<double>("minSCEtEndcapsPflow", 0.0);
+  desc.add<double>("minEOverPBarrelPflow", 0.0);
+  desc.add<double>("maxEOverPBarrelPflow", 999999999.0);
+  desc.add<double>("minEOverPEndcapsPflow", 0.0);
+  desc.add<double>("maxEOverPEndcapsPflow", 999999999.0);
+  desc.add<double>("maxDeltaEtaBarrelPflow", 999999999.0);
+  desc.add<double>("maxDeltaEtaEndcapsPflow", 999999999.0);
+  desc.add<double>("maxDeltaPhiBarrelPflow", 999999999.0);
+  desc.add<double>("maxDeltaPhiEndcapsPflow", 999999999.0);
+  desc.add<double>("hOverEConeSizePflow", 0.15);
+  desc.add<double>("hOverEPtMinPflow", 0.0);
+  desc.add<double>("maxHOverEBarrelPflow", 999999999.0);
+  desc.add<double>("maxHOverEEndcapsPflow", 999999999.0);
+  desc.add<double>("maxHBarrelPflow", 0.0);
+  desc.add<double>("maxHEndcapsPflow", 0.0);
+  desc.add<double>("maxSigmaIetaIetaBarrelPflow", 999999999.0);
+  desc.add<double>("maxSigmaIetaIetaEndcapsPflow", 999999999.0);
+  desc.add<double>("maxFbremBarrelPflow", 999999999.0);
+  desc.add<double>("maxFbremEndcapsPflow", 999999999.0);
+  desc.add<bool>("isBarrelPflow", false);
+  desc.add<bool>("isEndcapsPflow", false);
+  desc.add<bool>("isFiducialPflow", false);
+  desc.add<double>("maxTIPPflow", 999999999.0);
 
   // Corrections
-  desc.add<std::string>("superClusterErrorFunction","EcalClusterEnergyUncertaintyObjectSpecific") ;
-  desc.add<std::string>("crackCorrectionFunction","EcalClusterCrackCorrection") ;
- }
+  desc.add<std::string>("superClusterErrorFunction", "EcalClusterEnergyUncertaintyObjectSpecific");
+  desc.add<std::string>("crackCorrectionFunction", "EcalClusterCrackCorrection");
+
+  desc.add<bool>("ecalWeightsFromDB", true);
+  desc.add<std::vector<std::string>>("ecalRefinedRegressionWeightFiles", {}); // if not from DB. Otherwise, keep empty
```

Comment:
This comment should be better added to the produced cfi, see 
https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideConfigurationValidationAndHelp#Adding_Comments